File: tapeta/ExampleWallpaperService.java
package com.example.wojtek.wallpaper;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Matrix;
import android.graphics.Paint;
import android.os.Handler;
import android.service.wallpaper.WallpaperService;
import android.view.SurfaceHolder;
import android.view.SurfaceView;

/**
 * Created by Marcin on 2017-01-08.
 */

public class ExampleWallpaperService extends WallpaperService {

    private final Handler handler = new Handler();


    @Override
    public Engine onCreateEngine() {
        return new TheEngine();
    }

    public class TheEngine extends WallpaperService.Engine {

        private int circleY = 100;

        private final Runnable mDrawFrame = new Runnable() {
            public void run() {
                drawFrame();
            }
        };
        private boolean isVisible;

        private int currentFrame = 0;
        private int[] frames = {
//                R.drawable.easter_animated_gif_bunny_behind_basket_1,
//                R.drawable.easter_animated_gif_bunny_behind_basket_2,
//                R.drawable.easter_animated_gif_bunny_behind_basket_3,
//                R.drawable.easter_animated_gif_bunny_behind_basket_4,
//                R.drawable.easter_animated_gif_bunny_behind_basket_5,
//                R.drawable.easter_animated_gif_bunny_behind_basket_6,
//                R.drawable.easter_animated_gif_bunny_behind_basket_7
        };
        private Bitmap mainBitmap;
        private float scaleX;
        private float scaleY;

        private void drawFrame() {
            final SurfaceHolder holder = getSurfaceHolder();

            mainBitmap = BitmapFactory.decodeResource(getResources(), frames[currentFrame]);
//             -- Napis

            final Paint paint = new Paint();
            paint.setColor(0xff99ff99);
            paint.setAntiAlias(true);
            paint.setStrokeWidth(2);
            paint.setStrokeCap(Paint.Cap.ROUND);
            paint.setStyle(Paint.Style.STROKE);
            paint.setTextSize(80f);
            if (isVisible) {
                Canvas canvas = holder.lockCanvas();
                canvas.save();


                currentFrame++;
                if (currentFrame == 7){
                    currentFrame = 0;
                }
                Matrix matrix = new Matrix();
                canvas.scale(scaleX, scaleY);
//                canvas.drawBitmap(mainBitmap, matrix, null);

//                 Napis
                //wyrysuj coś prostego
                //zaysuj stare
                canvas.drawARGB(255,0,50,0);
                //canvas.getH
                circleY+=10;
                canvas.drawCircle(100,circleY,50,paint);
                canvas.drawText("FRANEK",100,circleY,paint);
                if(circleY>1000) circleY=100;

                canvas.restore();
                holder.unlockCanvasAndPost(canvas);
                //przygotowanie następnej ramki
                int frameDuration = 200;
                handler.postDelayed(mDrawFrame, frameDuration);
                //handler.post(mDrawFrame);
            }
        }

        @Override
        public void onVisibilityChanged(boolean visible) {
            this.isVisible = visible;
            if (isVisible) {
                handler.post(mDrawFrame);
            } else {
                handler.removeCallbacks(mDrawFrame);
            }
        }

        @Override
        public void onSurfaceChanged(SurfaceHolder holder, int format, int width, int height){
            super.onSurfaceChanged(holder, format, width, height);
            drawFrame();
            scaleX = width / (1f * mainBitmap.getWidth());
            scaleY = height / (1f * mainBitmap.getHeight());
        }

    }
}

File: app/src/main/java/com/example/wojtek/wallpaper/ExampleWallpaperService.kt
package com.example.wojtek.wallpaper

import android.graphics.Color
import android.graphics.Paint
import android.os.Handler
import android.service.wallpaper.WallpaperService

class ExampleWallpaperService : WallpaperService() {

    private val handler = Handler()

    override fun onCreateEngine() = TheEngine()

    inner class TheEngine : WallpaperService.Engine() {
        private val runnable = Runnable { drawFrame() }
        private val x = 540f
        private val textXStart = -600f
        private val yStart = 450
        private val eyeOffset = 20f
        private val eyeSize = 10f
        private val mouthSize = 7f


        private var y = yStart
        private var textX = textXStart
        private var visible: Boolean = false
        private var direction = 1

        private fun drawFrame() {
            y += 10 * direction
            textX += 10

            val holder = surfaceHolder
            val paint = preparePaint()
            if (visible) {
                val canvas = holder.lockCanvas()
                canvas.drawARGB(255, 0, 50, 0)
                canvas.drawCircle(x, y.toFloat(), 50f, paint)
                canvas.drawCircle(x - eyeOffset, y - 20f, eyeSize, blackPaint)
                canvas.drawCircle(x + eyeOffset, y - 20f, eyeSize, blackPaint)

                canvas.drawCircle(x, y + 130f, 100f, paint)
                canvas.drawCircle(x, y + 90f, eyeSize, blackPaint)
                canvas.drawCircle(x, y + 150f, eyeSize, blackPaint)

                canvas.drawCircle(x, y + 340f, 150f, paint)
                canvas.drawCircle(x, y + 290f, eyeSize, blackPaint)
                canvas.drawCircle(x, y + 340f, eyeSize, blackPaint)
                canvas.drawCircle(x, y + 390f, eyeSize, blackPaint)

                canvas.drawCircle(x - eyeOffset, y + 15f, mouthSize, blackPaint)
                canvas.drawCircle(x + eyeOffset, y + 15f, mouthSize, blackPaint)
                canvas.drawCircle(x - eyeOffset / 2, y + 22f, mouthSize, blackPaint)
                canvas.drawCircle(x + eyeOffset / 2, y + 22f, mouthSize, blackPaint)
                canvas.drawCircle(x, y + 25f, mouthSize, blackPaint)

                canvas.drawText("Wesołych świąt!", textX, 350f, paint)
                if (y !in yStart..1050)
                    direction *= -1
                if (textX > 1100)
                    textX = textXStart
                holder.unlockCanvasAndPost(canvas)
                handler.postDelayed(runnable, 20)
            }
        }

        override fun onVisibilityChanged(visible: Boolean) {
            this.visible = visible
            if (this.visible)
                handler.post(runnable)
            else
                handler.removeCallbacks(runnable)

        }
    }
}

private fun preparePaint(): Paint {
    val paint = Paint()
    paint.color = Color.WHITE
    paint.isAntiAlias = true
    paint.strokeWidth = 2f
    paint.strokeCap = Paint.Cap.ROUND
    paint.style = Paint.Style.FILL
    paint.textSize = 80f
    return paint
}

private val blackPaint: Paint by lazy {
    val paint = Paint()
    paint.color = Color.BLACK
    paint.isAntiAlias = true
    paint.strokeWidth = 2f
    paint.strokeCap = Paint.Cap.ROUND
    paint.style = Paint.Style.FILL
    paint.textSize = 80f
    paint
}